src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsFrameworkCompatibility.cs

```diff
@@ -43,17 +44,33 @@ public IEnumerable<FrameworkName> GetFrameworksSupportingNetStandard(FrameworkNa
             return CompatibilityListProvider
                 .Default
                 .GetFrameworksSupporting(nuGetFramework)
-                .Select(GetFrameworkName);
+                .Select(FrameworkNameUtility.GetFrameworkName);
         }
 
-        private NuGetFramework GetNuGetFramework(FrameworkName frameworkName)
+        public FrameworkName GetNearest(FrameworkName targetFramework, IEnumerable<FrameworkName> frameworks)
         {
-            return NuGetFramework.ParseFrameworkName(frameworkName.ToString(), DefaultFrameworkNameProvider.Instance);
-        }
+            if (targetFramework == null)
+            {
+                throw new ArgumentNullException(nameof(targetFramework));
+            }
 
-        private FrameworkName GetFrameworkName(NuGetFramework nuGetFramework)
-        {
-            return new FrameworkName(nuGetFramework.DotNetFrameworkName);
+            if (frameworks == null)
```

Comment:
The inner implementation accepts an empty list, always returning null. So it seems like a contrived restriction at this level. Also, this is something the caller can very easily throw based on (if he so desires). Could a reasonable compromise be to explicitly state in the XML docs that an empty list is accepted?